pkg/gitserver/client.go

```diff
@@ -529,6 +529,76 @@ func (c *Client) IsRepoCloned(ctx context.Context, repo api.RepoName) (bool, err
 	return cloned, nil
 }
 
+func (c *Client) AreReposCloned(ctx context.Context, repos ...api.RepoName) (*protocol.AreReposClonedResponse, error) {
+	numPossibleShards := len(c.Addrs(ctx))
+	shards := make(map[string]*protocol.AreReposClonedRequest, (len(repos)/numPossibleShards)*2) // 2x because it may not be a perfect division
```

Comment:
size of shards is maximum `len(c.Addrs)`. Normally it will be len(c.Addrs) since we should have a atleast one repo per gitserver replica.

```suggestion
	shards := make(map[string]*protocol.AreReposClonedRequest, numPossibleShards)
```